### PR TITLE
Add PlayerPassengerInputEvent

### PR DESCRIPTION
**Pull request by TheCalypso**
Hello everyone !

I am very happy to be able to present to you the little work that our team has done. Indeed this PR makes it possible to add the recovery of the inputs that a player makes when he is on an entity. To try to best meet expectations, we worked with @lynxplay .

Here is the link to the first discussion I had via my #7770 

As what has been said, we have done everything so that this event is done before the server logic. the location we have chosen allows us to better manage the event. 

We hope this PR will be a help for a potential future full entity control API. In the meantime, we thank you in advance for your feedback.

### CODE REVIEW

File: patches/server/0907-Add-PlayerPassengerInputEvent-call-in-handlePlayerIn.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -494,9 +494,16 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
+     @Override
+     public void handlePlayerInput(ServerboundPlayerInputPacket packet) {
+         PacketUtils.ensureRunningOnSameThread(packet, this, this.player.getLevel());
+-        this.player.setPlayerInput(packet.getXxa(), packet.getZza(), packet.isJumping(), packet.isShiftKeyDown());
++        // Paper start - create event control mounted entity
++        var event = new io.papermc.paper.event.player.PlayerPassengerInputEvent(player.getBukkitEntity(),
++                packet.getXxa(), packet.getZza(), packet.isJumping(), packet.isShiftKeyDown());
++        if (event.callEvent()) return;
```
Comment (lynxplay) [inline patches/server/0907-Add-PlayerPassengerInputEvent-call-in-handlePlayerIn.patch]: This should be `if(!event.callEvent())`. `Event#callEvent` returns `false` if the event was cancelled, e.g. we would want to return at that point.

**Review by lynxplay — CHANGES_REQUESTED** [review]

File: patches/server/0907-Add-PlayerPassengerInputEvent-call-in-handlePlayerIn.patch (hunk 1 above)
Comment (electronicboy) [reply patches/server/0907-Add-PlayerPassengerInputEvent-call-in-handlePlayerIn.patch]: Is this cancellation safe, i.e. it's not gonna cause a desync?

Comment (TheCalypso) [reply patches/server/0907-Add-PlayerPassengerInputEvent-call-in-handlePlayerIn.patch]: Oh thank you very much! Sorry I just fixed this :)

Comment (TheCalypso) [reply patches/server/0907-Add-PlayerPassengerInputEvent-call-in-handlePlayerIn.patch]: > Is this cancellation safe, i.e. it's not gonna cause a desync?

What do you mean by desync?

**Pull request closed without merging**